spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -241,7 +277,11 @@ def construct_spark_application(
         spark_app = self.compute_all_task_data(raw_data, spark_app)
         spark_app = self.compute_all_stage_data(raw_data, spark_app)
         spark_app = self.compute_all_driver_accum_data(raw_data, spark_app)
-        spark_app = self.compute_all_metadata(raw_data, spark_app)
+
+        metadata = self.compute_all_metadata(raw_data, spark_app)
```

Comment:
Small change to make this method not responsible for setting the `metadata` field but instead actually just computing it. We should probably do the same for the other methods as well, but I didn't do that in this PR

Reply:
Thank you!